@app/server/src/plugins/CreateUploadUrlPlugin.ts

```diff
@@ -0,0 +1,170 @@
+}
+
+/** The minimal set of information that this plugin needs to know about users. */
+interface User {
+  id: string;
+  isVerified: boolean;
+}
+
+async function getCurrentUser(pool: Pool): Promise<User | null> {
+  await pool.query("SAVEPOINT");
+  try {
+    const {
+      rows: [row],
+    } = await pool.query(
+      "select id, is_verified from app_public.users where id = app_public.current_user_id()"
+    );
+    if (!row) {
+      return null;
+    }
+    return {
+      id: row.id,
+      isVerified: row.is_verified,
+    };
+  } catch (err) {
+    await pool.query("ROLLBACK TO SAVEPOINT");
+    throw err;
+  } finally {
+    await pool.query("RELEASE SAVEPOINT");
+  }
+}
+/** The set of content types that we allow users to upload.*/
+const ALLOWED_UPLOAD_CONTENT_TYPES = [
+  "image/apng",
+  "image/bmp",
+  "image/gif",
+  "image/jpeg",
+  "image/png",
+  "image/svg+xml",
+  "image/tiff",
+  "image/webp",
+];
+
+const CreateUploadUrlPlugin = makeExtendSchemaPlugin(() => ({
+  typeDefs: gql`
+    """
+    All input for the \`createUploadUrl\` mutation.
+    """
+    input CreateUploadUrlInput @scope(isMutationInput: true) {
+      """
+      An arbitrary string value with no semantic meaning. Will be included in the
+      payload verbatim. May be used to track mutations by the client.
+      """
+      clientMutationId: String
+
+      """
+      You must provide the content type (or MIME type) of the content you intend
+      to upload. For further information about content types, see
+      https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types
+      """
+      contentType: String!
+    }
+
+    """
+    The output of our \`createUploadUrl\` mutation.
+    """
+    type CreateUploadUrlPayload @scope(isMutationPayload: true) {
+      """
+      The exact same \`clientMutationId\` that was provided in the mutation input,
+      unchanged and unused. May be used by a client to track mutations.
+      """
+      clientMutationId: String
+
+      """
+      Upload content to this signed URL.
+      """
+      uploadUrl: String!
+    }
+
+    extend type Mutation {
+      """
+      Get a signed URL for uploading files. It will expire in 5 minutes.
+      """
+      createUploadUrl(
+        """
+        The exclusive input argument for this mutation. An object type, make sure to see documentation for this object’s fields.
+        """
+        input: CreateUploadUrlInput!
+      ): CreateUploadUrlPayload
+    }
+  `,
+  resolvers: {
+    Mutation: {
+      async createUploadUrl(
+        _query,
+        args: { input: CreateUploadUrlInput },
+        context: OurGraphQLContext,
+        _resolveInfo
+      ) {
+        if (!uploadBucket) {
+          const err = new Error(
+            "Server misconfigured: missing `S3_UPLOADS_BUCKET` envvar"
+          );
+          // @ts-ignore
+          err.code = "MSCFG";
+          throw err;
+        }
+
+        const user = await getCurrentUser(context.rootPgPool);
```

Comment:
I believe this is what is causing the input error issue, changing `rootPgPool` to `pgClient` fixes it.